src/System.Text.Utf8String/System/Text/Utf8String.cs

```diff
@@ -587,6 +587,43 @@ public Utf8String TrimStart(byte[] trimCodeUnits)
             throw new NotImplementedException();
         }
 
+        public Utf8String TrimStart(Utf8String prefix)
```

Comment:
Can you please add a test like "abad".TrimStart("ab") returns "d" to make it explicit that we are intentionally following the same behavior as string.TrimStart() and not going with option a)?